plugins/woocommerce/src/Blocks/Utils/ProductCollectionUtils.php

```diff
@@ -92,4 +99,171 @@ private static function remove_empty_array_recursive( $array ) {
 		}
 		return $array;
 	}
+
+	/**
+	 * Parse editor's location context from WP Post.
+	 *
+	 * Possible contexts:
+	 * - post
+	 * - page
+	 * - product
+	 * - product-archive
+	 * - cart
+	 * - checkout
+	 * - catalog
+	 * - order
+	 *
+	 * @param WP_Post $post The Post instance.
+	 *
+	 * @return string Returns the context.
+	 */
+	public static function parse_editor_location_context( $post ) {
+		$context = 'site';
+
+		if ( ! $post instanceof \WP_Post ) {
+			return $context;
+		}
+
+		$post_type = $post->post_type;
+		if ( ! in_array( $post_type, array( 'post', 'page', 'wp_template', 'wp_template_part' ), true ) ) {
+			return $context;
+		}
+
+		if ( in_array( $post_type, array( 'wp_template', 'wp_template_part' ), true ) ) {
+
+			$name = $post->post_name;
+			if ( false !== strpos( $name, SingleProductTemplate::SLUG ) ) {
+				$context = 'product';
+			} elseif ( ProductAttributeTemplate::SLUG === $name ) {
+				$context = 'product-archive';
+			} elseif ( false !== strpos( $name, 'taxonomy-' ) ) { // Including the '-' in the check to avoid false positives.
+				$taxonomy           = str_replace( 'taxonomy-', '', $name );
+				$product_taxonomies = get_object_taxonomies( 'product', 'names' );
+				if ( in_array( $taxonomy, $product_taxonomies, true ) ) {
+					$context = 'product-archive';
+				}
+			} elseif ( in_array( $name, array( CartTemplate::SLUG, MiniCartTemplate::SLUG ), true ) ) {
+				$context = 'cart';
+			} elseif ( CheckoutTemplate::SLUG === $name ) {
+				$context = 'checkout';
+			} elseif ( ProductCatalogTemplate::SLUG === $name ) {
+				$context = 'catalog';
```

Comment:
In the editor I used `product-catalog`. Worth unifying.

```suggestion
				$context = 'product-catalog';
```